java-client/src/main/java/org/apache/pegasus/security/KerberosProtocol.java

```diff
@@ -50,19 +52,21 @@ class KerberosProtocol implements AuthProtocol {
   // request. The JAAS framework defines the term "subject" to represent the source of a request. A
   // subject may be any entity, such as a person or a service.
   private Subject subject;
-  private String serviceName;
-  private String serviceFqdn;
-  private String keyTab;
-  private String principal;
+  private final String serviceName;
+  private final String serviceFqdn;
+  private final String keyTab;
+  private final String principal;
   final int CHECK_TGT_INTEVAL_SECONDS = 10;
   final ScheduledExecutorService service =
       Executors.newSingleThreadScheduledExecutor(
           new ThreadFactory() {
             @Override
             public Thread newThread(Runnable r) {
-              Thread t = new Thread(r);
+              String timestamp = new SimpleDateFormat("HHmmss").format(new Date());
```

Comment:
from JDK8 , advise migration to the [java.time](http://docs.oracle.com/javase/8/docs/api/java/time/package-summary.html) classes.
String timestamp = LocalTime.now().format(DateTimeFormatter.ofPattern("HHmmss"));

Reply:
> from JDK8 , advise migration to the [java.time](http://docs.oracle.com/javase/8/docs/api/java/time/package-summary.html) classes. String timestamp = LocalTime.now().format(DateTimeFormatter.ofPattern("HHmmss"));

ok, i have changed to use java.time.